### PR TITLE
FEAT: adds macOS GUI backend

### PR DESCRIPTION
**Pull request by qtxie**

### CODE REVIEW

File: modules/view/backends/platform.red [hunk 1]
```diff
@@ -232,6 +265,10 @@ system/view/platform: context [
 			_italic:		symbol/make "italic"
 			_underline:		symbol/make "underline"
 			_strike:		symbol/make "strike"
+			_border:		symbol/make "border"
+			_backdrop:		symbol/make "backdrop"
+			_font-name:		symbol/make "font-name"
+			_font-size:		symbol/make "font-size"
```
Comment (dockimbel) [inline modules/view/backends/platform.red]: Why those symbols are needed?

File: system/formats/Mach-APP.r [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		either Windows? [
+			rejoin [{"} to-local-file file {"}]
+		][
+			to-local-file file
+		]
+	]
+	
+	run: func [cmd [string!]][
+		trim/lines cmd
+		either verbose [?? cmd call/console cmd][call/wait cmd]
```
Comment (dockimbel) [inline system/formats/Mach-APP.r]: Debugging code leftover.

File: system/formats/Mach-APP.r [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	]
+	
+	copy-files: func [src [file!] dst [file!]][	
+		run reform [
+			either Windows? ["xcopy"]["cp -R"]
+			to-OS-file src
+			to-OS-file dst
+			either Windows? ["/Y /E /I"][""]
```
Comment (dockimbel) [inline system/formats/Mach-APP.r]: Remove Windows-specific codes.

File: system/assets/osx/Info.plist [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>CFBundleExecutable</key>
+	<string>$Red-App-Name$</string>
+	<key>CFBundleIconFile</key>
+	<string>AppIcon</string>
+	<key>CFBundleIdentifier</key>
+	<string>com.redlang.$Red-App-Name$</string>
```
Comment (dockimbel) [inline system/assets/osx/Info.plist]: org.redlang....

File: system/utils/libRedRT-exports.r [hunk 5]
```diff
@@ -321,6 +322,7 @@
 	red/natives/unset*
 	red/natives/handle-thrown-error
 	red/natives/now*
+	red/natives/get-env*
```
Comment (dockimbel) [inline system/utils/libRedRT-exports.r]: Add also `list-env*`.

File: modules/view/backends/platform.red [hunk 6]
```diff
@@ -327,6 +381,8 @@ system/view/platform: context [
 			][
 				as red-value! switch evt/type [
 					EVT_TIME		 [_time]
+					EVT_DRAW		 [_draw]
```
Comment (dockimbel) [inline modules/view/backends/platform.red]: It should be renamed "redraw".

File: modules/view/backends/platform.red [hunk 7]
```diff
@@ -510,14 +616,27 @@ system/view/platform: context [
...
 
-	request-font: routine [font [object!] mono? [logic!]][
-		gui/OS-request-font font mono?
+	exit-event-loop: routine [][
+		#switch OS [
+			Windows  [gui/PostQuitMessage 0]
+			MacOSX   [gui/evt-loop-cnt: gui/evt-loop-cnt - 1]
+			#default [0]
+		]
+	]
```
Comment (dockimbel) [inline modules/view/backends/platform.red]: Should be moved to console code.

File: system/formats/Mach-APP.r (hunk 3 above)
Comment (qtxie) [inline system/formats/Mach-APP.r]: It's used for cross compilation.

**Pull request closed without merging**